cmd/clusterctl/client/repository/template.go

```diff
@@ -136,3 +139,52 @@ func NewTemplate(input TemplateInput) (Template, error) {
 		objs:            objs,
 	}, nil
 }
+
+// MergeTemplates merges the provided Templates into one Template.
+// Notes on the merge operation:
+// - The merge operation returns an error if all the templates do not have the same TargetNamespace.
+// - The Variables of the resulting template is a union of all Variables in the templates.
+// - The default value is picked from the first temmplate that defines it.
+//    The defaults of the same variable in the subsequent templates will be ignored.
```

Comment:
This makes sense, if we put this into a context
```suggestion
// - The default value is picked from the first temmplate that defines it.
//    The defaults of the same variable in the subsequent templates will be ignored.
//    (e.g when merging a cluster template and its ClusterClass, the default value from the template takes precedence)
```